### ISSUE TITLE
handleSubmit callback automatically called

### ISSUE THREAD

**Issue by iphonic** [erikras/redux-form#3374]
Hi, I have been using redux-form for a while, I am facing a very strange behavior. I am adding a few lines of my code here from **render()** function of my component.

```
render(){

    const {handleSubmit, invalid, pristine, submitting } = this.props;

    const styles = reactCSS({
      'default': {
        color: {
          width: '40px',
          height: '20px',
          border: '5px',
          background: `rgba(${ this.state.color.r }, ${ this.state.color.g }, ${ this.state.color.b }, ${ this.state.color.a })`,
        },
        swatch: {
          padding: '6px',
          background: '#fff',
          borderRadius: '1px',
          boxShadow: '0 0 0 1px rgba(0,0,0,.1)',
          display: 'inline-block',
          cursor: 'pointer',
          margin: '0px 0px 0px 35px',
          float: 'left',
          position: 'relative'
        },
        popover: {
          position: 'absolute',
          zIndex: '2',
        },
        cover: {
          position: 'fixed',
          top: '0px',
          right: '0px',
          bottom: '0px',
          left: '0px',
        },
      },
    });

    return(
        <form onSubmit={handleSubmit(this.onFormSubmit.bind(this))}>
                <Field
                  name="color"
                  label="Booth Color"
                  placeholder="Choose booth color"
                  type="text"
                  mandatory="true"
                  component={this.renderColorTextField}
                />
                <div style={ styles.swatch }>
                  <button style={styles.color} onClick={ this.handlePickerClick.bind(this) }></button>
                </div>
                  { this.state.displayColorPicker ? <div style={ styles.popover }>
                    <div style={ styles.cover } onClick={ this.handlePickerClose.bind(this) }/>
                    <ColorPicker color={ this.state.color } onChange={ this.handlePickerChange.bind(this) } />
                  </div> : null }
              <div className="booth-footer">
                  <div className="col-md-6">
                    <Button className="button-theme map-button booth-button button-theme-blue"  disabled={ pristine || invalid || submitting } loading={this.state.loading} type="submit">Save</Button>
                  </div>
                  <div className="col-md-6">
                    <Button className="button-theme map-button booth-button" onClick={this.onCancel.bind(this)}>Cancel</Button>
                  </div>
              </div>
        </form>
      </div>
    );
}

handlePickerChange(color){
    this.setState({ color: color.rgb});
    this.props.change('color',color.hex);
    //Update the color of the booth as it changes
    var updatedBooth={...this.props.booth,color:color.hex};
    this.props.updateBoothColor(updatedBooth);
}

handlePickerClick(){
    this.setState({ displayColorPicker: !this.state.displayColorPicker });
}

onFormSubmit(values){
    if(this.props.submitting){
        this.setState({'loading':true});
    }
}
```

In the above code, I have a read-only Field which is getting updated with HEX value picked from ColorPicker.. the strange problem is whenever I click on ColorPicker to pick color the **onFormSubmit** callback is getting called on its own without clicking on Submit button, I have to handle  onFormSubmit by using **this.props.submitting** property of redux-form, which is successfully blocking unwanted callback process, but when I submitting **this.props.submitting** never gets true and I am not able to submit...

**Summing up I have two problems**

**1. handleSubmit callback is automatically called which shouldn't be the case**
**2. if 1. is caused by something else how to determine if user actually click Submit button to submit the values?**

Any solution will help, please help.

Thanks.

**Comment by gustavohenke** [comment]
Button tags are, by default, `type="submit"`, which is a W3C Spec, and is not related to Redux Form or React.

That's a very common error among all web developers, AFAIC 😄

**Comment by iphonic** [comment]
@gustavohenke Thanks for reply, but this is not related to button **type="submit"** the problem is redux-form calling **handleSubmit** on its own without clicking on submit button, and clicking on submit button not calling handleSubmit handler.. how can changing type of button would solve it?

**Comment by iphonic** [comment]
Thanks @gustavohenke it did work, I added type="button" to all the buttons in form..
